rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/converter/CourseOfStudyConverter.java

```diff
@@ -25,36 +23,33 @@
 
     @Override
     public CourseOfStudy createFrom(final AddCourseOfStudyRequest dto) {
-        Optional<Faculty> opFaculty = facultyRepository.findFacultyByName(dto.getFaculty());
-        Optional<SystemUser> opUser1 = userRepository.findByLogin(dto.getContactPerson1());
-        Optional<SystemUser> opUser2 = userRepository.findByLogin(dto.getContactPerson2());
-        CourseType type = dto.getCourseType().equals("FULL_TIME") ? CourseType.FULL_TIME : CourseType.EXTERNAL;
-
-        if (opFaculty.isPresent() && opUser1.isPresent()) {
-            // Check if Contact Person 2 was given but is missing in DB
-            if (!dto.getContactPerson2().equals("") && opUser2.isEmpty()) {
-                String msg = String.format("User %s not found!", dto.getContactPerson2());
-                throw new UserNotFoundException(msg);
-            }
-
-            return CourseOfStudy.builder()
-                    .name(dto.getName())
-                    .faculty(opFaculty.get())
-                    .courseType(type)
-                    .contactPerson1(opUser1.get())
-                    .contactPerson2(opUser2.orElse(null))
-                    .isJoined(dto.getIsJoined())
-                    .remarks(dto.getRemarks())
-                    .build();
+        Faculty opFaculty = facultyRepository.findFacultyByName(dto.getFaculty())
+                .orElseThrow(() ->
+                        new FacultyNotFoundException(String.format("Faculty %s not found!", dto.getFaculty())));
+
+        SystemUser opUser1 = userRepository.findByLogin(dto.getContactPerson1())
+                .orElseThrow(() ->
+                        new UserNotFoundException(String.format("User %s not found!", dto.getContactPerson1())));
+        SystemUser opUser2;
+        String opUser2Login = dto.getContactPerson2();
+        if (opUser2Login != null) {
+            opUser2 = null;
```

Comment:
I think it should be `if (opUser2Login == null) {...`

Reply:
Fixed

Reply:
Done